internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/RequestTopicForPushRequest.java

```diff
@@ -0,0 +1,178 @@
+package com.linkedin.venice.controllerapi;
+
+import com.linkedin.venice.meta.Version.PushType;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class RequestTopicForPushRequest {
+  private final String clusterName;
+  private final String storeName;
+  private final PushType pushType;
+  private final String pushJobId;
+
+  private boolean sendStartOfPush = false;
+  private boolean sorted = false; // an inefficient but safe default
+  private boolean isWriteComputeEnabled = false;
+  private long rewindTimeInSecondsOverride = -1L;
+  private boolean deferVersionSwap = false;
+  private String targetedRegions = null;
+  private int repushSourceVersion = -1;
+  private Set<String> partitioners = Collections.emptySet();
+  private String compressionDictionary = null;
+  private X509Certificate certificateInRequest = null;
+  private String sourceGridFabric = null;
+  private String emergencySourceRegion = null;
+
+  public RequestTopicForPushRequest(String clusterName, String storeName, PushType pushType, String pushJobId) {
+    if (clusterName == null || clusterName.isEmpty()) {
+      throw new IllegalArgumentException("clusterName is required");
+    }
+    if (storeName == null || storeName.isEmpty()) {
+      throw new IllegalArgumentException("storeName is required");
+    }
+    if (pushType == null) {
+      throw new IllegalArgumentException("pushType is required");
+    }
+
+    if (pushJobId == null || pushJobId.isEmpty()) {
+      throw new IllegalArgumentException("pushJobId is required");
+    }
+
+    this.clusterName = clusterName;
+    this.storeName = storeName;
+    this.pushType = pushType;
+    this.pushJobId = pushJobId;
+  }
+
+  public static PushType extractPushType(String pushTypeString) {
```

Comment:
In the PushType enum itself, we already have an valueOf (int), would it be better to add this method there to have all the valueOf in one place? 

And this method is simply catch an illegalArgumentException and throw a new illegalArgumentException with a more informative error message. Catching exception is expensive. So maybe inside the PushType enum, you can pre-build a static map of pushType string, if the inputString does not match any, you can throw an illegalArgumentException. 

Just my two-cents.

Reply:
I agree with you. I've updated code to use pre-built map.